File: Formula1/bin/Formula1/F4.php
<?php
namespace Formula1;

class F4 extends Monoplaza {
    private $paisCategoria;

     ////////CONSTRUCTOR////////
    public function __construct(string $nombrePiloto, string $nacionalidad, int $numeroPlaza, string $escuderia, int $cantidadPuntos, string $paisCategoria) {
        parent::__construct($nombrePiloto, $nacionalidad, $numeroPlaza, $escuderia, $cantidadPuntos);
        $this->paisCategoria = $paisCategoria;
    }

     // GETTERS Y SETTERS ////
    public function getPaisCategoria(){
    return $this->paisCategoria;
    }
    public function setPaisCategoria($paisCategoria){
    $this->paisCategoria = $paisCategoria;
    }

     // METODOS ///
    public function posicionValida($posicion){
        if ($posicion < 1) {
        return false;
    }
    if ($posicion > 30) {
        return false;
    }
    return true;
    }

    public function otorgarPuntos($posicion,$vueltaRapida){
        if ($posicion < 1 || $posicion > 30) {
        return false;
        }
        switch ($posicion) {
        case 1:
            $this->cantidadPuntos += 25;
            break;
        case 2:
            $this->cantidadPuntos += 18;
            break;
        case 3:
            $this->cantidadPuntos += 15;
            break;
        case 4:
            $this->cantidadPuntos += 12;
            break;
        case 5:
            $this->cantidadPuntos += 10;
            break;
        case 6:
            $this->cantidadPuntos += 8;
            break;
        case 7:
            $this->cantidadPuntos += 6;
            break;
        case 8:
            $this->cantidadPuntos += 4;
            break;
        case 9:
            $this->cantidadPuntos += 2;
            break;
        case 10:
            $this->cantidadPuntos += 1;
        default:
            break;
        }
    }

    public function subirCategoria(string $nombreAcademia){
        return new F3($this->nombrePiloto, $this->nacionalidad, $this->numeroPlaza, $this->escuderia, $this->cantidadPuntos, $nombreAcademia);
    }

    ///TOSTRING///
    public function __toString(){
    return parent::__toString() . "País Categoría:" . $this->paisCategoria;
    }
}

File: Formula1/src/Formula1/JAVA/F4.java
package Formula1.JAVA;

public class F4 extends Monoplaza {
    private String paisCategoria;

    // Constructor
    public F4(String nombrePiloto, String nacionalidad, int numeroPlaza, String escuderia, int cantidadPuntos, String paisCategoria) {
        super(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos);
        this.paisCategoria = paisCategoria;
    }

    // Getters y setters
    public String getPaisCategoria() {
        return paisCategoria;
    }

    public void setPaisCategoria(String paisCategoria) {
        this.paisCategoria = paisCategoria;
    }

    //////METODOS/////////
    
    @Override
    public boolean posicionValida(int posicion) {
        if(posicion>=1 && posicion<=30){
            return true;
        }else{
            return false;
        }
    }
    
    @Override
    public void otorgarPuntos(int posicion, boolean vueltaRapida) {
        if (!posicionValida(posicion)) return;

        switch (posicion) {
            case 1 -> cantidadPuntos += 25;
            case 2 -> cantidadPuntos += 18;
            case 3 -> cantidadPuntos += 15;
            case 4 -> cantidadPuntos += 12;
            case 5 -> cantidadPuntos += 10;
            case 6 -> cantidadPuntos += 8;
            case 7 -> cantidadPuntos += 6;
            case 8 -> cantidadPuntos += 4;
            case 9 -> cantidadPuntos += 2;
            case 10 -> cantidadPuntos += 1;
            default -> cantidadPuntos += 0;
        }
    }

    public F3 subirCategoria(String nombreAcademia) {
    return new F3(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos,nombreAcademia);
    }
    @Override
    public String toString() {
    return super.toString() + " | País Categoría: " + paisCategoria;
    }
}
